ocaml/util/xapi_version.ml

```diff
@@ -18,28 +18,31 @@ let hostname = "localhost"
 
 let date = Xapi_build_info.date
 
+let parse_xapi_version version =
+  try Scanf.sscanf version "%d.%d.%s" (fun maj min rest -> (maj, min, rest))
```

Comment:
We have an `Rpm` module in XAPI which has a `Pkg.parse_epoch_version_release` and various comparison functions for it. In particular a version may contain more than just major and minor and the actual RPM version rules can be quite tricky.
E.g. in a backport we might have to increment (or introduce) a patch version in addition to major and minor in case the major.minor+1 is already taken up by an existing release.

Reply:
The xapi version string is in format "major.minor.git_id", i.e.: "23.25.0", "0.0.dev" or "22.18.0.11.g93d35f038", I checked related `Rpm` code: I think it is different from `Rpm.epoch_ver_rel` as there is no "epoch" and "release" info inside "xapi_version".

Reply:
I don't think Rpm can be used here. The versions compared here are runtime ones, not rpm ones